core/src/nucleus/ribosome/api/commit.rs

```diff
@@ -25,71 +25,74 @@ pub fn invoke_commit(
 ) -> Result<Option<RuntimeValue>, Trap> {
     // deserialize args
     let args_str = runtime_args_to_utf8(&runtime, &args);
-    let res_entry: Result<CommitArgs, _> = serde_json::from_str(&args_str);
-    // Exit on error
-    if res_entry.is_err() {
-        // Return Error code in i32 format
-        return Ok(Some(RuntimeValue::I32(
-            HcApiReturnCode::ErrorSerdeJson as i32,
-        )));
-    }
+    let entry_input: CommitArgs = match serde_json::from_str(&args_str) {
+        Ok(entry_input) => entry_input,
+        // Exit on error
+        Err(_) => {
+            // Return Error code in i32 format
+            return Ok(Some(RuntimeValue::I32(
+                HcApiReturnCode::ErrorSerdeJson as i32,
+            )));
+        }
+    };
```

Comment:
Unfortunately we can't use `.or()?` here because we want to return an `Ok` and `?` only returns `Err`s.